### PR TITLE
WEBAPP-262: Use api v3 categories endpoint

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://integreat.atlassian.net/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/modules/common/containers/LanguageSelector.js [hunk 1]
```diff
@@ -33,19 +32,19 @@ export class LanguageSelector extends React.Component {
    */
   getLanguageChangeAction (languageCode) {
```
Comment (Schedulaar) [inline src/modules/common/containers/LanguageSelector.js]: This doesn't return an action anymore, right?

File: src/modules/endpoint/endpoints/__tests__/categories.spec.js [hunk 2]
```diff
@@ -8,104 +8,78 @@ jest.unmock('../categories')
...
     id: 3649,
-    permalink: {
-      url_page: 'willkommen'
-    },
+    path: '/augsburg/de/willkommen',
     title: 'Willkommen',
     status: 'publish',
     content: '',
-    parent: 0,
+    parent: {},
```
Comment (Schedulaar) [inline src/modules/endpoint/endpoints/__tests__/categories.spec.js]: why do we need an object here? 🤔

File: src/modules/endpoint/models/CategoryModel.js [hunk 3]
```diff
@@ -4,25 +4,23 @@ import type { Moment } from 'moment'
...
-    this._url = params.url
+    this._path = params.path
     this._title = params.title
     this._content = params.content
-    this._parentId = params.parentId
-    this._parentUrl = params.parentUrl
+    this._parentPath = params.parentPath
     this._thumbnail = params.thumbnail
     this._order = params.order
     this._path = params.path
```
Comment (Schedulaar) [inline src/modules/endpoint/models/CategoryModel.js]: that's a duplicate

File: src/modules/endpoint/models/CategoryModel.js [hunk 4]
```diff
@@ -4,25 +4,23 @@ import type { Moment } from 'moment'
...
-  _parentUrl: string
+  _parentPath: string
   _thumbnail: string
   _order: number
   _availableLanguages: Map<string, string>
   _lastUpdate: Moment
 
-  constructor (params: {| id: number, path: string, url: string, title: string, content: string, parentId: number, thumbnail: string,
-    parentUrl: string, order: number, availableLanguages: Map<string, string>, lastUpdate: Moment |}) {
+  constructor (params: {| id: number, path: string, path: string, title: string, content: string, thumbnail: string,
```
Comment (Schedulaar) [inline src/modules/endpoint/models/CategoryModel.js]: `path` is duplicated

File: src/routes/categories/components/__tests__/CategoryListItem.spec.js [hunk 5]
```diff
@@ -40,7 +40,7 @@ describe('CategoryListItem', () => {
   })
 
   it('should replace empty thumbnail', () => {
-    const noThumbCategory = {id: 42, url: '', title: 'GotNoThumb :O', parentId: 2}
+    const noThumbCategory = {id: 42, path: '', title: 'GotNoThumb :O', parentId: 2}
```
Comment (Schedulaar) [inline src/routes/categories/components/__tests__/CategoryListItem.spec.js]: We don't need a parentId anymore

File: src/routes/categories/containers/__tests__/CategoriesPage.spec.js [hunk 6]
```diff
@@ -12,21 +12,21 @@ describe('CategoriesPage', () => {
...
       thumbnail: 'no_thumbnail',
-      parentUrl: ''
+      parentPath: ''
     }), new CategoryModel({
       id: 3650,
-      url: '/augsburg/de/anlaufstellen',
+      path: '/augsburg/de/anlaufstellen',
       title: 'Anlaufstellen zu sonstigen Themen',
       content: '',
       parentId: 0,
```
Comment (Schedulaar) [inline src/routes/categories/containers/__tests__/CategoriesPage.spec.js]: Maybe also adjust the test-set here as well, so we don't have parentId

File: src/routes/categories/containers/__tests__/__snapshots__/CategoriesToolbar.spec.js.snap [hunk 7]
```diff
@@ -10,7 +10,7 @@ exports[`CategoriesToolbar should render Toolbar, if category can be found 1`] =
 </Toolbar>
 `;
 
-exports[`CategoriesToolbar should render root-url for pdf endpoint 1`] = `
+exports[`CategoriesToolbar should render root-path for pdf endpoint 1`] = `
```
Comment (Schedulaar) [inline src/routes/categories/containers/__tests__/__snapshots__/CategoriesToolbar.spec.js.snap]: Here its a URL though :D

File: src/modules/common/containers/LanguageSelector.js (hunk 1 above)
Comment (klinzo) [reply src/modules/common/containers/LanguageSelector.js]: yes it still does (with exception of categories pathname), but I will adjust this (and change the name) to always return a  path with WEBAPP-231.

File: src/modules/endpoint/endpoints/__tests__/categories.spec.js (hunk 2 above)
Comment (klinzo) [reply src/modules/endpoint/endpoints/__tests__/categories.spec.js]: This is the json returned by the api v3, which returns the id, the full cms url and the path for each parent, so parent is an object.

File: src/modules/endpoint/models/CategoryModel.js (hunk 3 above)
Comment (klinzo) [reply src/modules/endpoint/models/CategoryModel.js]: ups :D

File: src/routes/categories/containers/__tests__/__snapshots__/CategoriesToolbar.spec.js.snap (hunk 7 above)
Comment (klinzo) [reply src/routes/categories/containers/__tests__/__snapshots__/CategoriesToolbar.spec.js.snap]: Changed

File: src/modules/endpoint/endpoints/__tests__/categories.spec.js (hunk 2 above)
Comment (Schedulaar) [reply src/modules/endpoint/endpoints/__tests__/categories.spec.js]: Ah I see... But maybe you could still fill them up to real parent objects like we get them from cms (for categories with no parent we get`{id: 0,  path: null, url: null }`)

File: src/modules/endpoint/models/CategoryModel.js (hunk 4 above)
Comment (klinzo) [reply src/modules/endpoint/models/CategoryModel.js]: Removed

File: src/modules/endpoint/endpoints/__tests__/categories.spec.js (hunk 2 above)
Comment (klinzo) [reply src/modules/endpoint/endpoints/__tests__/categories.spec.js]: okay, sure :)

**Pull request merged**